### ISSUE TITLE
Keyboard navigation on date picker doesn't work until up or down arrow key is pressed

### ISSUE THREAD

**Issue by OshanAbeykoon** [Hacker0x01/react-datepicker#2261]
**Bug**
The way how keyboard navigation works is fine for a date picker like follows, As the right and left arrow keys will traverse through the input field but arrow up or down makes focus to the selected date and from there on wards the keyboard navigation works as intended. 
![image](https://user-images.githubusercontent.com/38980527/87301121-a4108380-c52c-11ea-8c70-ad540bd49b1f.png)

But this can be an issue for a date picker with **portal version**. 
Issue 1: When the date picker is opened (in portal version) the user has to explicitly press up or down arrow **first** before pressing left or right arrow keys. 
This can be re created on the demo page. 

Issue 2:  Pressing left or right arrow buttons on the date picker header element also makes the keyboard navigation not to work.
Recreation step: Open the date picker (in portal version) - > Press down arrow - > Press left arrow (Keyboard navigation works at this point) - > Press left or right arrow button - > Press any keyboard arrow key  (Nothing happens, keyboard navigation stopped working)
![image](https://user-images.githubusercontent.com/38980527/87303111-39614700-c530-11ea-81fd-18f06056004c.png)

**Expected behavior**

1. When the date picker is opened with **portal version**, a user should be able to press any arrow key to navigate through dates.

2. Keyboard navigation should continue to work after interacting with date picker header elements.
